app/components/announcement_banner/index.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
+import withObservables from '@nozbe/with-observables';
+import {of as of$, combineLatest} from 'rxjs';
+import {switchMap} from 'rxjs/operators';
+
+import {observeConfigBooleanValue, observeConfigValue, observeLastDismissedAnnouncement, observeLicense} from '@queries/servers/system';
+
+import AnnouncementBanner from './announcement_banner';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+
+const enhanced = withObservables([], ({database}: WithDatabaseArgs) => {
+    const lastDismissed = observeLastDismissedAnnouncement(database);
+    const bannerText = observeConfigValue(database, 'BannerText');
+    const allowBannerDismissal = observeConfigBooleanValue(database, 'AllowBannerDismissal');
+
+    const bannerDismissed = combineLatest([lastDismissed, bannerText, allowBannerDismissal]).pipe(
+        switchMap(([ld, bt, abd]) => of$(abd && (ld === bt))),
+    );
+
+    const license = observeLicense(database);
+    const enableBannerConfig = observeConfigBooleanValue(database, 'EnableBanner');
+    const bannerEnabled = combineLatest([license, enableBannerConfig]).pipe(
```

Comment:
is the announcement banner a EE feature? are we sure?

Reply:
The logic was on V1 like that, and I needed to add the license to set it.

Reply:
@esethna Could you confirm? Or who is the owner of this feature?